lib/moonshot/stack_parameter.rb

```diff
@@ -0,0 +1,64 @@
+module Moonshot
+  class StackParameter
+    attr_reader :name
+    attr_reader :default
+
+    def initialize(name, default: nil, use_previous: false)
+      @name = name
+      @default = default
+      @use_previous = use_previous
+      @value = nil
+    end
+
+    # Does this Stack Parameter have a default value that will be used?
+    def default?
+      !@default.nil?
+    end
+
+    def use_previous?
+      @use_previous ? true : false
+    end
+
+    # Has the user provided a value for this parameter?
+    def set?
+      !@value.nil?
```

Comment:
This makes me think if an empty string is considered okay as a value for a parameter.